### PR TITLE
Add JS file to decrypt with AES-GCM

### PR DESCRIPTION
**Pull request by elijahsoria**
Adds a decryption method that uses subtle.crypto.Decrypt to decrypt with AES-GCM, supported on all browsers.

### CODE REVIEW

File: js/aes_gcm.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+/**
```
Comment (dvoytenko) [inline js/aes_gcm.js]: Add license header.

File: js/aes_gcm.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  return array;
+}
+
+/**
+ * Decrypts the input text using AES-GCM with the input key.
+ * @param {string} key
+ * @param {string} text
+ * @return {!Promise}
+ */
+export function decryptAesGcm(key, text) {
```
Comment (dvoytenko) [inline js/aes_gcm.js]: This should be the first function at the very top.

File: js/aes_gcm.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
+/**
+ * Subtle-based AES-GCM decryption supported on all browser types.
+ */
+
+/** 
+ * Possibly wrap an object in a Promise.
```
Comment (dvoytenko) [inline js/aes_gcm.js]: "Converts IE11 CryptoOperation type to a Promise. See https://msdn.microsoft.com/en-us/windows/dn280996%28v=vs.71%29"

File: js/aes_gcm.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+ * @param {Object} op
+ * @return {!Promise}
+ */
+function wrapCryptoOp(op) {
+  if (typeof op.then == 'function') {
+    return op;
+  }
+  return new Promise(function (resolve, reject) {
```
Comment (dvoytenko) [inline js/aes_gcm.js]: No space between "function" and "(". And use "=>" as much as possible.

File: js/aes_gcm.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    op.oncomplete = function (e) {
+      resolve(op.result);
+    };
+    op.onerror = function (e) {
```
Comment (dvoytenko) [inline js/aes_gcm.js]: `op.onerror = reject;`

File: js/aes_gcm.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+ * @param {string} str
+ * @return {!Uint8Array}
+ */
+function base64Decode(str) {
+  const bytes = atob(str);
+  const len = bytes.length;
+  const array = new Uint8Array(len);
+  for (let i = 0; i < len; i++) {
+    const charCode = bytes.charCodeAt(i);
+    if (charCode >= 256) {
```
Comment (dvoytenko) [inline js/aes_gcm.js]: Let's drop this `if`

File: js/aes_gcm.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+/**
+ * Decrypts the input text using AES-GCM with the input key.
+ * @param {string} key
+ * @param {string} text
+ * @return {!Promise}
+ */
+export function decryptAesGcm(key, text) {
+  const keybytes = base64Decode(key);
+  const isIE = !!window.msCrypto;
+  const subtle = (window.crypto || window.msCrypto).subtle;
```
Comment (dvoytenko) [inline js/aes_gcm.js]: Let's replace `window` with `self` everywhere here. This way we get a worker-ready code.

File: js/aes_gcm.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+          },
+          formattedkey,
+          // IE requires "tag" to be removed from the bytes.
+          isIE ? bytesToDecrypt.slice(0, bytesToDecrypt.byteLength - 16) : bytesToDecrypt
+        ))
+        .then(function (buffer) {
+          // 5. Decryption gives us raw bytes and we need to turn them into text.
+          const decryptedBytes = new Uint8Array(buffer);
+          return utf8Decode(decryptedBytes);
+        }, function (error) {
```
Comment (dvoytenko) [inline js/aes_gcm.js]: Remove the catch function from here.

File: js/aes_gcm.js [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+            // Edge requires "tagLength".
+            tagLength: 128 // block size (16): 1-128
+          },
+          formattedkey,
+          // IE requires "tag" to be removed from the bytes.
+          isIE ? bytesToDecrypt.slice(0, bytesToDecrypt.byteLength - 16) : bytesToDecrypt
+        ))
+        .then(function (buffer) {
+          // 5. Decryption gives us raw bytes and we need to turn them into text.
+          const decryptedBytes = new Uint8Array(buffer);
```
Comment (dvoytenko) [inline js/aes_gcm.js]: Combine these two lines

File: js/aes_gcm.js [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+          return utf8Decode(decryptedBytes);
+        }, function (error) {
+          throw error;
+        });
+    });
+}
```
Comment (dvoytenko) [inline js/aes_gcm.js]: nit: new line

File: js/aes_gcm.js (hunk 3 above)
Comment (elijahsoria) [reply js/aes_gcm.js]: Thanks!

File: js/aes_gcm.js (hunk 4 above)
Comment (elijahsoria) [reply js/aes_gcm.js]: Done.

File: js/aes_gcm.js (hunk 7 above)
Comment (elijahsoria) [reply js/aes_gcm.js]: Done.

File: js/aes_gcm.js [hunk 11]
```diff
@@ -0,0 +1,104 @@
+/* Copyright 2019 The Subscribe with Google Authors. All Rights Reserved.
```
Comment (dvoytenko) [inline js/aes_gcm.js]: This should be:

```
/**
 * Copyright
```

I know it's annoying, but I believe this matters for some transforming tools.

Comment (elijahsoria) [reply js/aes_gcm.js]: Done.

**Pull request merged**